docs/src/main/sphinx/connector/teradata.md

```diff
@@ -0,0 +1,334 @@
+connection-url=jdbc:teradata://example.teradata.com/
+connection-user=***
+connection-password=***
+```
+
+The connection-url defines the connection information and parameters to pass to the Teradata JDBC driver. The supported parameters for the URL are available in the Teradata JDBC documentation.
+
+For example, the following connection-url configures connection pooling and character encoding:
+```example
+connection-url=jdbc:teradata://example.teradata.com/CHARSET=UTF8,TMODE=ANSI
+```
+The connection-user and connection-password are typically required and determine the user credentials for the connection, often a service user.
+
+ClearScape Analytics configuration
+For ClearScape Analytics environments, you can configure additional properties:
+```clearscape properties
+connector.name=teradata
+connection-url=jdbc:teradata://clearscape-host.teradata.com/
+connection-user=demo_user
+connection-password=***
+teradata.clearscape.url=https://api.clearscape.teradata.com
+teradata.clearscape.region=asia-south
+```
+## Connection security
+If you have TLS configured with a globally-trusted certificate installed on your data source, you can enable TLS between your cluster and the data source by appending parameters to the JDBC connection string set in the connection-url catalog configuration property.
+
+For example, to enable SSL encryption:
+```
+connection-url=jdbc:teradata://example.teradata.com/ENCRYPTDATA=ON
+```
+For more information on security configuration options, see the Teradata JDBC security documentation.
+
+
+## Multiple Teradata databases
+You can have as many catalogs as you need, so if you have additional Teradata databases, simply add another properties file to etc/catalog with a different name, making sure it ends in .properties. For example, if you name the property file sales.properties, Trino creates a catalog named sales using the configured connector.
+
+
+## Table properties
+Table property usage example:
+```
+CREATE TABLE person (
+  id INTEGER NOT NULL,
+  name VARCHAR(100),
+  age INTEGER,
+  birthday DATE
+)
+WITH (
+  primary_index = ARRAY['id'],
+  table_kind = 'SET'
+);
+```
+The following are supported Teradata table properties:
+
+* - Property name
+  - Required
+  - Description
+* - primary_index
+  - No
+  - The primary index of the table. Can specify multiple columns.
+* - table_kind
+  - No
+  - Specifies whether the table is a SET (default) or MULTISET table.
+* - fallback
+  - No
+  - Enables or disables fallback protection. Values: true, false.
+## Type mapping
+Because Trino and Teradata each support types that the other does not, this connector modifies some types <type-mapping-overview> when reading or writing data. Data types may not map the same way in both directions between Trino and the data source.
+
+
+### Teradata to Trino type mapping
+The connector maps Teradata types to the corresponding Trino types following this table:
+
+
+* - Teradata database type
+  - Trino type
+
+* - BYTEINT
```

Comment:
https://github.com/Teradata/trino/blob/e4db4d22deaa446ea5739ac08642327514df6f78/plugin/trino-teradata/src/main/java/io/trino/plugin/teradata/TeradataClient.java#L960 Please refer this code to document Data Type Mapping for read operations. Seems mapping documentation is not aligned with implementation.